### ISSUE TITLE
Quit content and Shut down options

### ISSUE THREAD

**Issue by justanickname** [libretro/Lakka#113]
Quit content option that makes Lakka quit the current game/emulator/video. It could be placed in the content menu (where Save/Load state, Reset and Resume options are). Also, perhaps a warning could pop up when the user tries to load something else:

"To load this content, Lakka will have to quit the content that is currently running. All unsaved progress will be lost. Do you wish to proceed?"

Shut down would let the user shut down his Lakka box cleanly without having to press the power button.

**Comment by ProfessorKaos64** [comment]
Second this

**Comment by twinaphex** [comment]
Quit Content is a good idea.

Shut down is an option that would not be as scalable and would therefore need to be optionally implemented based on the host device. For instance, Apple on iOS does not allow apps to be closed by user interaction, never mind allow the device to be shutdown by user interaction, and there are similar issues like that on other devices. So it will need to be properly abstracted away and implemented in such a way that thr option will only appear on systems that allow for this.

TLDR - it can be considered but it will take some work.

**Comment by floreal** [comment]
@twinaphex, I think @justanickname was just talking about it in Lakka distro. Maybe allowing to customize menu to add shell commands would help.

**Comment by twinaphex** [comment]
No, that is not going to scale well at all and it would be a huge security risk. Proper solutions are going to be implemented here and theyre going to be implemented in RetroArch.

**Comment by floreal** [comment]
Maybe with an option « --enable-shutdow-device » at compile time ?

**Comment by twinaphex** [comment]
I'll implement something for this later this week.

**Comment by ProfessorKaos64** [comment]
How does XBMC/Kodi handle this? That contains effective shutdown options if I recall right, across many distros/platforms.

**Comment by Kivutar** [comment]
Kodi doesn't try to work on iOS/PSP/XBOX/etc, let @twinaphex think about the best implementation for what RetroArch is and the payload of platforms it supports.

**Comment by Kivutar** [comment]
Done by a patch for now.
